tests/unit/test_charm.py

```diff
@@ -1,66 +1,426 @@
+        self.create_n2_relation()
+
+        self.harness.update_config(key_values={})
+        self.harness.evaluate_status()
+
+        assert self.harness.charm.unit.status == WaitingStatus(
+            "Waiting for statefulset to be patched"
+        )
+
+    def test_give_storage_is_not_attached_when_config_changed_then_status_is_waiting(self):
+        test_statefulset = StatefulSet(
+            spec=StatefulSetSpec(
+                selector=LabelSelector(),
+                serviceName="whatever",
+                template=PodTemplateSpec(
+                    spec=PodSpec(
+                        containers=[
+                            Container(
+                                name=WORKLOAD_CONTAINER_NAME,
+                                securityContext=SecurityContext(privileged=True),
+                            )
+                        ]
+                    )
+                ),
+            )
+        )
+        self.mock_lightkube_client_get.return_value = test_statefulset
+        self.mock_check_output.return_value = b"1.1.1.1"
+        self.create_n2_relation()
+
+        self.harness.update_config(key_values={})
+        self.harness.evaluate_status()
+
+        assert self.harness.charm.unit.status == WaitingStatus(
+            "Waiting for storage to be attached"
+        )
+
+    def test_given_amf_n2_information_is_not_available_when_config_changed_then_status_is_waiting(
+        self,
+    ):
+        test_statefulset = StatefulSet(
+            spec=StatefulSetSpec(
+                selector=LabelSelector(),
+                serviceName="whatever",
+                template=PodTemplateSpec(
+                    spec=PodSpec(
+                        containers=[
+                            Container(
+                                name=WORKLOAD_CONTAINER_NAME,
+                                securityContext=SecurityContext(privileged=True),
+                            )
+                        ]
+                    )
+                ),
+            )
+        )
+        self.mock_lightkube_client_get.return_value = test_statefulset
+        self.mock_check_output.return_value = b"1.1.1.1"
+        self.harness.add_storage("config", attach=True)
+        self.create_n2_relation()
+
+        self.harness.update_config(key_values={})
+        self.harness.evaluate_status()
 
-    def test_config_changed(self):
-        self.assertEqual(list(self.harness.charm._stored.things), [])
-        self.harness.update_config({"thing": "foo"})
-        self.assertEqual(list(self.harness.charm._stored.things), ["foo"])
+        assert self.harness.charm.unit.status == WaitingStatus("Waiting for N2 information")
 
-    def test_action(self):
-        # the harness doesn't (yet!) help much with actions themselves
-        action_event = Mock(params={"fail": ""})
-        self.harness.charm._on_fortune_action(action_event)
+    def test_given_all_status_check_are_ok_when_config_changed_then_status_is_active(self):
```

Comment:
Could we  also check the situation that self.mock_check_output does not return a pod ip ?

Reply:
There is such test. The name is `test_given_pod_ip_is_not_available_when_config_changed_then_status_is_waiting` (line 111).

Reply:
OK, sorry for missing this.